### ISSUE TITLE
Can't build lib with exceptions disabled

### ISSUE THREAD

**Issue by alphanso** [odygrd/quill#357]
Hi,

I am trying to migrate to v3.3.1. I am unable to build with exceptions disabled.

Thanks

P.S It's not happening with 2.9.0. Bug is also reproducible with 2.9.2

**Comment by odygrd** [comment]
hey sorry for the late response. What operating system is it ? And what compiler and version ?
There is a linux build with no exceptions on CI that is passing https://github.com/odygrd/quill/actions/runs/6713304824/job/18244606128

**Comment by odygrd** [comment]
Disabling exceptions on windows is not supported at the moment

**Comment by harlan-merlin** [comment]
Same issue here and can confirm 2.9.0 works. Glancing at the difference between the two, looks like [`check_printf_format_string(...)`](https://github.com/odygrd/quill/blob/master/quill/include/quill/detail/misc/Common.h#L178) was added between then and now.

Pre-Build Steps:
```
wget https://github.com/odygrd/quill/archive/refs/tags/v3.4.0.tar.gz
tar -xvf v3.4.0.tar.gz 
mkdir build && cd build
```

CMake
```
cmake -DQUILL_NO_EXCEPTIONS:BOOL=ON -DQUILL_BUILD_EXAMPLES=OFF ..
-- The C compiler identification is AppleClang 15.0.0.15000040
-- The CXX compiler identification is AppleClang 15.0.0.15000040
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- CMAKE_BUILD_TYPE: Release
-- QUILL_VERSION: 3.4.0
-- QUILL_NO_EXCEPTIONS: ON
-- QUILL_FMT_EXTERNAL: OFF
-- QUILL_NO_THREAD_NAME_SUPPORT: OFF
-- Configuring done (0.7s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/harlan/GitHub/quill-3.4.0/build
```

Make
```
make
[  4%] Building CXX object quill/CMakeFiles/quill.dir/src/detail/backend/BackendWorker.cpp.o
In file included from /Users/harlan/GitHub/quill-3.4.0/quill/src/detail/backend/BackendWorker.cpp:1:
In file included from /Users/harlan/GitHub/quill-3.4.0/quill/include/quill/detail/backend/BackendWorker.h:10:
In file included from /Users/harlan/GitHub/quill-3.4.0/quill/include/quill/Config.h:9:
/Users/harlan/GitHub/quill-3.4.0/quill/include/quill/detail/misc/Common.h:194:5: fatal error: cannot use 'throw' with exceptions disabled
    throw std::runtime_error{
    ^
1 error generated.
make[2]: *** [quill/CMakeFiles/quill.dir/src/detail/backend/BackendWorker.cpp.o] Error 1
make[1]: *** [quill/CMakeFiles/quill.dir/all] Error 2
make: *** [all] Error 2

% clang --version
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin22.6.0
Thread model: posix
```

**Comment by odygrd** [comment]
thanks for reporting :) that is fixed in ed934b50fbe0b2729670f42d19cb5fc745cd177e
Please let me know if you have any other issues
